### PR TITLE
More features towards completion

### PR DESCRIPTION
**Pull request by matanlurey**
Closes https://github.com/dart-lang/code_builder/issues/90
Closes https://github.com/dart-lang/code_builder/issues/92
Closes https://github.com/dart-lang/code_builder/issues/93

### CODE REVIEW

File: lib/src/builders/type/type_def.dart [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+        type?.buildType(scope),
+      );
+    }).toList(),
+    $closeBracket,
+  );
+}
+
+/// Lazily build a `typedef`, or [FunctionTypeAlias] AST.
+///
+///
```
Comment (natebosch) [inline lib/src/builders/type/type_def.dart]: [nit] delete 2 blank doc comment lines

Comment (matanlurey) [inline lib/src/builders/type/type_def.dart]: Done.

**Comment by zoechi** [comment]
beta.6 is not yet published. Is this an oversight?

**Comment by matanlurey** [comment]
Wasn't a priority immediately but I can publish some time this weekend. I
really should look into an auto publish Travis script of some sort.

**Comment by zoechi** [comment]
Just wanted to know. 
I could get rid of a few ugly hacks with the things you added here. 
If nothing specific is blocking publishing, it would be nice.

**Pull request merged**